cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -117,6 +118,54 @@ struct ARROW_EXPORT AzureOptions {
   MakeDataLakeServiceClient() const;
 };
 
+namespace internal {
+
+enum class HNSSupport {
```

Comment:
Again I don't feel strongly, but I would not have abbreviated so much. Yes, the docstring explains in detail but someone who just sees that name and Googles it will find themselves learning about the [Croatian Football Federation](https://en.wikipedia.org/wiki/Croatian_Football_Federation) :smile:

Reply:
I abbreviated the enum class because it gets repeated a lot, but kept the function names with the full name :-)

HNS is used in Microsoft documentation together with SMB, FTP... so it's not so bad.

https://learn.microsoft.com/en-us/azure/storage/blobs/storage-feature-support-in-storage-accounts

Reply:
I would rather have the full name as well, even if unfortunately long.

Reply:
Can I at least alias it in `azurefs.cc`?